.github/workflows/cleanup-self-hosted-runners.yml

```diff
@@ -0,0 +1,59 @@
+name: Cleanup Azure self hosted runners
+run-name: Cleanup Azure self hosted runners
+
+on:
+  schedule:
+    # Run every 6 hours
+    - cron: "0 */6 * * *"
+  workflow_dispatch:
+
+# The following secrets are required for this workflow to run:
+# AZURE_CREDENTIALS - Credentials for the Azure CLI. It's recommended to set up a resource
+#                     group specifically for self-hosted Actions Runners.
+#   az ad sp create-for-rbac --name "{YOUR_DESCRIPTIVE_NAME_HERE}" --role contributor \
+#     --scopes /subscriptions/{SUBSCRIPTION_ID_HERE}/resourceGroups/{RESOURCE_GROUP_HERE} \
+#     --sdk-auth
+# AZURE_RESOURCE_GROUP - Resource group to create the runner(s) in
+jobs:
+  delete-runner:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Azure Login
+      uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+    - name: Discover VMs to delete
+      uses: azure/CLI@v1
+      with:
+        azcliversion: 2.54.0
```

Comment:
We'll probably want to use the [latest](https://learn.microsoft.com/en-us/cli/azure/release-notes-azure-cli) 2.64.0, now that it's a few months later :)

Reply:
Okay!

Reply:
Should we do the same with the `create-azure-self-hosted-runners` and `delete-self-hosted-runner` GitHub workflows, too? They're at v2.43.0...

Reply:
Good catch! Updated those here: https://github.com/git-for-windows/git-for-windows-automation/pull/94

Another thing I just noticed that this here PR uses `azure/CLI@v1`, while we're using `azure/CLI@v2` in the other pipelines. Another thing to update :)